mmv1/third_party/terraform/fwvalidators/framework_validators.go

```diff
@@ -115,3 +117,109 @@ func (v nonEmptyStringValidator) ValidateString(ctx context.Context, request val
 func NonEmptyStringValidator() validator.String {
 	return nonEmptyStringValidator{}
 }
+
+func StringSet(d basetypes.SetValue) []string {
+
+	StringSlice := make([]string, 0)
+	for _, v := range d.Elements() {
+		StringSlice = append(StringSlice, v.(basetypes.StringValue).ValueString())
+	}
+	return StringSlice
+}
+
+// Define the possible service account name patterns
+var serviceAccountNamePatterns = []string{
+	`^.+@.+\.iam\.gserviceaccount\.com$`,                     // Standard IAM service account
+	`^.+@developer\.gserviceaccount\.com$`,                   // Legacy developer service account
+	`^.+@appspot\.gserviceaccount\.com$`,                     // App Engine service account
+	`^.+@cloudservices\.gserviceaccount\.com$`,               // Google Cloud services service account
+	`^.+@cloudbuild\.gserviceaccount\.com$`,                  // Cloud Build service account
+	`^service-[0-9]+@.+-compute\.iam\.gserviceaccount\.com$`, // Compute Engine service account
+}
+
+// Create a custom validator for service account names
+type ServiceAccountNameValidator struct{}
```

Comment:
nit: isn't this validator validating emails?

Reply:
changed name to `ServiceAccountEmailValidator